web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -10,24 +10,24 @@ odoo.define('web_app_drawer.app_drawer', function(require) {
     var core = require('web.core');
     
     Menu.include({
-        
+        // Force all_outside to prevent app icons from going into more menu
         reflow: function() {
             this._super('all_outside');
         },
-        
     });
     
     SearchView.include({
         
-        // ReImplement core toggle_visibility to not focus when mobile
+        // Prevent focus of search field on mobile devices
         toggle_visibility: function (is_visible) {
-            this.do_toggle(!this.headless && is_visible);
-            if (this.$buttons) {
-                this.$buttons.toggle(!this.headless && is_visible && this.visible_filters);
-            }
-            if (!this.headless && is_visible && !this.isMobile()) {
-                this.$('div.oe_searchview_input').last().focus();
-            }
+            var self = this,
+                $searchField = $('div.oe_searchview_input').last();
+            $searchField.one('focus', function(){
```

Comment:
You might want to use a proxy instead, to have more extendibility.

That could be by adding a method like this to the object:

```
unfocus_in_mobiles: function (event) {
    if (this.isMobile()) {
        event.preventDefault();
    }
}
```

And then proxying it in this line:

```
this.$('div.oe_searchview_input').one('focus', $.proxy(this.unfocus_in_mobiles, this));
```

It's a good habit to structure your JS modules that way. It allows it to be more extensible by submodules, and avoids `self=this` hacks by proxying (which is actually how JS was designed to be, be it crazy or not).